tests/test_gear_yaml2cfg.py

```diff
@@ -97,3 +113,20 @@ def test_yaml2cfg_test_no_header():
         shallow=False,
         )
     p.unlink()
+
+
+def test_yaml_duplicated_params(default_yaml_files):
```

Comment:
This does not solve #911 and becomes unecessary if the schema validation is implemented, best to do that instead of this which seems like half measure

Reply:
took me some time to truly understand what you wanted, but I finally managed to pull out some kind of logic that make sense.